From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Mon, 4 Jan 2021 16:06:43 +0100
Subject: Fix 32bit sign comparisons

sieve-binary.c: In function 'sieve_binary_get_resource_usage':
sieve-binary.c:199:54: warning: comparison of integer expressions of different signedness: 'time_t' {aka 'long int'} and 'unsigned int' [-Wsign-compare]
  199 |  if (update_time != 0 && (ioloop_time - update_time) > timeout)
      |                                                      ^
---
 pigeonhole/src/lib-sieve/sieve-binary.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pigeonhole/src/lib-sieve/sieve-binary.c b/pigeonhole/src/lib-sieve/sieve-binary.c
index 06cf598..c971921 100644
--- a/pigeonhole/src/lib-sieve/sieve-binary.c
+++ b/pigeonhole/src/lib-sieve/sieve-binary.c
@@ -196,7 +196,7 @@ void sieve_binary_get_resource_usage(struct sieve_binary *sbin,
 	time_t update_time = header->resource_usage.update_time;
 	unsigned int timeout = sbin->svinst->resource_usage_timeout_secs;
 
-	if (update_time != 0 && (ioloop_time - update_time) > timeout)
+	if (update_time != 0 && (ioloop_time - update_time) > (time_t)timeout)
 		i_zero(&header->resource_usage);
 
 	sieve_resource_usage_init(rusage_r);
